crates/biome_js_analyze/src/semantic_analyzers/correctness/no_const_assign.rs

```diff
@@ -113,4 +116,58 @@ impl Rule for NoConstAssign {
             ),
         )
     }
+
+    fn action(ctx: &RuleContext<Self>, _: &Self::State) -> Option<JsRuleAction> {
+        let node = ctx.query();
+        let model = ctx.model();
+        let mut mutation = ctx.root().begin();
+
+        let declared_binding = model.binding(node)?;
+
+        if let Some(possible_declarator) = declared_binding.syntax().ancestors().find(|node| {
+            !AnyJsObjectBindingPatternMember::can_cast(node.kind())
+                && !JsObjectBindingPatternPropertyList::can_cast(node.kind())
+                && !AnyJsArrayBindingPatternElement::can_cast(node.kind())
+                && !JsArrayBindingPatternElementList::can_cast(node.kind())
+                && !JsIdentifierBinding::can_cast(node.kind())
+        }) {
+            if JsVariableDeclarator::can_cast(possible_declarator.kind()) {
+                let possible_declaration = possible_declarator.parent()?;
+                if let Some(js_variable_declaration) =
+                    JsVariableDeclaratorList::cast_ref(&possible_declaration)
+                        .and_then(|declaration| declaration.syntax().parent())
+                        .and_then(JsVariableDeclaration::cast)
```

Comment:
You could use something like:

```rs
match possible_declarator.declaration()? {
    AnyJsVariableDeclaration::JsVariableDeclaration(js_variable_declaration) => {}
    AnyJsVariableDeclaration::JsForVariableDeclaration(js_for_variable_declaration) => {}
}
```

Or even better, you can directly retrieve the kind token from the declaration:

```rs
let declaration = possible_declarator.declaration()?;
debug_assert!(declaration.is_const(), "The declaration should be const.");
let const_token = declaration.kind_token();
...
```

Reply:
Same method is not available.

Reply:
I think you have to import `biome_js_syntax::stmt_ext::JsVariableDeclarator`